Create issue templates

jamcat22 opened the pull request:
This PR adds issue templates for requests to add, update, and remove site listings.

I tried to incorporate the points made in both the existing issues linked below.

Let me know if y'all have any feedback on the changes!

Closes #4309 and closes #2318.

jamcat22 committed:
Create issue templates

Adds templates for requests to add, update, and remove site listings.

Closes #4309 and #2318.

phallobst commented on .github/ISSUE_TEMPLATE/update-existing-site.md:
```diff
@@ -0,0 +1,43 @@
+- [ ] Its Twitter username, Facebook page, and/or email address have changed.
+- [ ] Its logo has changed.
+- [ ] It has been acquired or absorbed by another site, and the new site is eligible to be listed on twofactorauth.org.
+- [ ] Other (please describe below).
```
Maybe have `It fits better into a different category.` (or something like that) as well? Or is it too rare an occasion to include it here?

jamcat22 replied:
Didn't even think of that! I've added it. 👍

jamcat22 committed:
Create issue template config

Add link to report a code of conduct violation.

jamcat22 committed:
Update update-existing-site.md

Add category site entry update option.
Updated title to include a descriptive icon.

jamcat22 committed:
Update config.yml

jamcat22 committed:
Update remove-existing-site.md

Add descriptive icon.

jamcat22 committed:
Update add-site-with-2fa.md

Add descriptive icon to title.

jamcat22 committed:
Update add-site-without-2fa.md

Add descriptive icon to title.

jamcat22 committed:
Update issue templates

Number templates. Add feature, bug, and blank templates. Modify template config to remove GitHub native blank issues.

jamcat22 committed:
Move issue template prefixes to filename

GitHub sorts issue templates by file name.

jamcat22 committed:
Fix checkboxes on Add site with 2FA template

jamcat22 committed:
Add checkbox markdown instructions

jamcat22 committed:
Update Report a Bug template

Quick sentence fix.

phallobst commented on .github/ISSUE_TEMPLATE/1-add-site-with-2fa.md:
```diff
@@ -0,0 +1,62 @@
+
+* Hardware:
+  - [ ] U2F keys
+  - [ ] WebAuthn
```
```suggestion
```
Does it make sense to distinguish "WebAuthn" from others if we don't have a dedicated `tfa` value for it?
If it's just for future enhancements, we could also include `biometric` then (and map it to `hardware`).

jamcat22 replied:
I'm guessing the original idea behind including it as a method in the template was to allow people to mark either U2F or WebAuthn and have us map both to `U2F` at the moment. We could also change it to "U2F/WebAuthn" or "FIDO2/U2F".

phallobst replied:
I'm afraid the users that use an issue template (as opposed to a pull request) are more likely _not_ the most tech-savvy ones, i.e. they might struggle to distinguish these in the first place.
Also, WebAuthn doesn't enforce a hardware solution, although most uses will probably include a hardware dongle of some sort.
In short: I'd go for a combined entry `FIDO2 / U2F keys`.

Carlgo11 replied:
Doesn’t FIDO2 include non-2FA authentication?

jamcat22 replied:
@phallobst, I agree on that front. I think matching the option to what a service typically would state in their documentation is the best way to go.

For example, [Dropbox](https://help.dropbox.com/teams-admins/team-member/enable-two-step-verification) states it as 

>A security key is a small USB, Bluetooth, or Near Field Communication (NFC) device that follows one of the open standards:
>‘FIDO Universal 2nd Factor (U2F)'
>'Web Authentication (WebAuthn)', also known as 'FIDO2’

and [Google](https://support.google.com/accounts/answer/6103523) calls it

> a security key that plugs into a USB port and works with FIDO Universal 2nd Factor (U2F) or FIDO2

so by listing the option as "FIDO2 / U2F security keys", I think we'd capture most of the terms used in end-user facing documentation.

Additionally, we could also update the linked wiki entry to show screenshots of what users would encounter when using a FIDO2/U2F key in their browser.

@carlgo11, while FIDO2/WebAuthn can be used in a passwordless state (either with the device acting as a "strong single factor" and needing no additional factor, or with the device requiring a user-created PIN to be entered before authenticating, similar to a smartcard) it can also be used as a second factor.

phallobst commented on .github/ISSUE_TEMPLATE/1-add-site-with-2fa.md:
```diff
@@ -0,0 +1,62 @@
+
+### Information about the site's eligibility to be added to twofactorauth.org: ###
+<!-- Check each box below to verify that the site meets our requirements for being listed.
+If a site does not meet any of these requirements, feel free to continue your issue submission.
```
```suggestion
If a site does not meet all of these requirements, feel free to continue your issue submission.
```

phallobst commented on .github/ISSUE_TEMPLATE/2-add-site-without-2fa.md:
```diff
@@ -0,0 +1,60 @@
+
+### Information about the site's eligibility to be added to twofactorauth.org: ###
+<!-- Check each box below to verify that the site meets our requirements for being listed.
+If a site does not meet any of these requirements, feel free to continue your issue submission.
```
```suggestion
If a site does not meet all of these requirements, feel free to continue your issue submission.
```

phallobst commented on .github/ISSUE_TEMPLATE/3-update-existing-site.md:
```diff
@@ -0,0 +1,49 @@
+<!-- List the specific changes to the site entry that should be made below.
+For example, if a site now supports 2FA, list the methods of 2FA the site now supports.
+Please include any supporting documentation for the changes as well, such as official documentation links or announcement blog posts.
+If no public-facing documentation is available, please include screenshots of the changes. -->
```
```suggestion
If no public-facing documentation is available, please include screenshots of the changes, redacting any personal information. -->
```

jamcat22 committed:
Apply suggestions from code review

Clarify submission requirements and ask that personal information be redacted from all requested screenshots.

Co-Authored-By: Michael Loßin <bugs@michael-lossin.de>

jamcat22 committed:
Update Report a Bug

Ask that personal information be redacted from any provided screenshots.

jamcat22 committed:
Update Add Site with 2FA issue template

Clarify FIDO2/U2F hardware method.
